hazelcast-client/src/main/java/com/hazelcast/client/spi/ClientProxyDescriptor.java

```diff
@@ -0,0 +1,26 @@
+/*
+ * Copyright (c) 2008-2016, Hazelcast, Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.hazelcast.client.spi;
+
+/***
+ * Descriptor of Hazelcast client proxies
+ */
+public interface ClientProxyDescriptor {
+    String getServiceName();
+
+    Class<? extends ClientProxy> getClientProxy();
```

Comment:
Would it not make more sense to call this method getClientProxyClass since a class is returned, not a proxy instance.

Perhaps the name can even be shortened, since client is already implied by this interface. So getProxyClass.